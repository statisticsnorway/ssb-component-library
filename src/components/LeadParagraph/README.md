Lead Paragraph
========

> A component for showing a lead paragraph
### Usage


#### HTML

```html
<p class="ssb-lead-paragraph">Default text here</p>

<p class="ssb-lead-paragraph negative">Negative text here</p>
```

#### React

```jsx harmony
<LeadParagraph>Default text here</LeadParagraph>

<LeadParagraph negative>Negative text here</LeadParagraph>
```

Available props:

| Name       | Type           | Description  |
| ---------- | ------------- | ----- |
| children   | Required node | Content within the wrapper |
| negative | Optional boolean | Styling for dark theme |
