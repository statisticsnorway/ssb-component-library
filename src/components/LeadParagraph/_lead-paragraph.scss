.ssb-lead-paragraph {
  @include open-sans;
  color: $ssb-dark-6;
  font-size: 20px;
  line-height: 1.7;

  &.negative {
    color: $ssb-white;
  }
}
