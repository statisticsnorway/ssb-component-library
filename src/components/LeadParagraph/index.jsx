import React from 'react';
import PropTypes from 'prop-types';

const LeadParagraph = ({ children, negative }) => <p className={negative ? 'negative' : ''}>{ children }</p>;

LeadParagraph.propTypes = {
	children: PropTypes.node,
	negative: PropTypes.bool,
};

export default LeadParagraph;
