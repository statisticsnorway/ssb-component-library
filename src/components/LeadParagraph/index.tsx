import React, { type ReactNode } from 'react'
import classNames from '../../utils/utils'

interface LeadParagraphProps {
  children?: ReactNode
  className?: string
  negative?: boolean
}

const LeadParagraph: React.FC<LeadParagraphProps> = ({ children, className, negative = false }) => (
  <p className={classNames('ssb-lead-paragraph', negative && 'negative', className)}>{children}</p>
)

export default LeadParagraph
