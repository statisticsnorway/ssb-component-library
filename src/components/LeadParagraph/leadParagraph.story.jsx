import React from 'react'
import LeadParagraph from './index'

export default {
  title: 'Lead Paragraph',
}

export const Standard = () => (
  <LeadParagraph>A lead paragraph is useful for making text stand out and be more significant.</LeadParagraph>
)
