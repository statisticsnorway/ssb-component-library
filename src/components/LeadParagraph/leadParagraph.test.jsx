import React from 'react';
import renderer from 'react-test-renderer';
import LeadParagraph from './index';

describe('LeadParagraph component', () => {
	test('Matches the snapshot', () => {
		const component = renderer.create(<LeadParagraph>Leading paragraph</LeadParagraph>).toJSON();
		expect(component).toMatchSnapshot();
	});
});
