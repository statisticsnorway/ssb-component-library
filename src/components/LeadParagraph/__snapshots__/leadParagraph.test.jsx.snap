// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LeadParagraph component Matches the snapshot 1`] = `
<p
  className=""
>
  Leading paragraph
</p>
`;
